pdata/pcommon/value.go

```diff
@@ -452,6 +452,33 @@ func (v Value) AsRaw() any {
 	return fmt.Sprintf("<Unknown OpenTelemetry value type %q>", v.Type())
 }
 
+func (v Value) Equal(c Value) bool {
+	if v.Type() != c.Type() {
+		return false
+	}
+
+	switch v.Type() {
+	case ValueTypeEmpty:
+		return v.Type() == c.Type()
```

Comment:
Already checked, just return true.

Reply:
Hmm no, the type check at the beginning of the method checks that the types are not the same (and therefore returns false).
Here, we check that if a is empty and b is empty, they are equal.